src/common/history.c

```diff
@@ -97,8 +97,11 @@ void dt_history_delete_on_image_ext(int32_t imgid, gboolean undo)
   dt_image_reset_final_size(imgid);
 
   /* remove darktable|style|* tags */
-  dt_tag_detach_by_string("darktable|style%", imgid);
```

Comment:
It seems to me that we want to always detach the tags here as it was done.

Reply:
The point here is to do not call the detach, that's why I'm not calling the attach. If no tag is attached there's no need to detach it.

Reply:
But this is a routine to delete history, what about old edits?

Reply:
As I see it is user responsability, if someone don't want the old tags when disabling this functionality is easy to delete them on the lighttable.